components/user/UserProfile.js

```diff
@@ -86,22 +89,57 @@ function UserProfile({ BASE_URL, data }) {
       )}
 
       {/* Passed Ref object as the ref attribute to the JSX of the DOM node of QR */}
-      <div className="flex justify-center my-4" ref={qrRef}>
-        {qrShow && (
-          <QRCodeCanvas
-            className="border border-white"
-            value={`${BASE_URL}/${data.username}`}
-            size={fallbackImageSize * 2}
-          />
-        )}
-      </div>
-      <div className="flex justify-center mb-4">
-        {qrShow && (
-          <Button primary={true} onClick={downloadQR}>
-            Download QR code
-          </Button>
-        )}
-      </div>
+      <Modal show={qrShow} setShow={setQrShow} modalStyles="w-fit m-auto">
+        <div className="flex flex-col items-center justify-center px-8">
+          <div>
+            <div className="flex justify-center my-4" ref={qrRef}>
+              {qrShow && (
+                <QRCodeCanvas
+                  className="border border-white"
+                  value={`${BASE_URL}/${data.username}`}
+                  size={fallbackImageSize * 2}
+                />
+              )}
+            </div>
+            <div className="w-[100%] px-2 mx-auto flex justify-center mb-4">
```

Comment:
Same here, `w-full` would be more consistent 

```suggestion
            <div className="w-full px-2 mx-auto flex justify-center mb-4">
```